presto-main/src/test/java/io/prestosql/server/security/oauth2/TestingHydraService.java

```diff
@@ -0,0 +1,183 @@
+    private static final SecureRandom SECURE_RANDOM = new SecureRandom();
+
+    private final Network network = Network.newNetwork();
+
+    private final PostgreSQLContainer<?> databaseContainer = new PostgreSQLContainer<>()
+            .withNetwork(network)
+            .withNetworkAliases("database")
+            .withUsername("hydra")
+            .withPassword("mysecretpassword")
+            .withDatabaseName("hydra");
+
+    private final GenericContainer<?> migrationContainer = createHydraContainer()
+            .withCommand("migrate sql --yes " + DSN)
+            .withStartupCheckStrategy(new OneShotStartupCheckStrategy().withTimeout(ofMinutes(5)));
+
+    private final FixedHostPortGenericContainer<?> consentContainer = new FixedHostPortGenericContainer<>("oryd/hydra-login-consent-node:v1.4.2")
+            .withNetwork(network)
+            .withNetworkAliases("consent")
+            .withExposedPorts(3000)
+            .withEnv("HYDRA_ADMIN_URL", "http://hydra:4445")
+            .withEnv("NODE_TLS_REJECT_UNAUTHORIZED", "0")
+            .waitingFor(Wait.forHttp("/").forStatusCode(200));
+
+    private final FixedHostPortGenericContainer<?> hydraContainer = createHydraContainer()
+            .withNetworkAliases("hydra")
+            .withExposedPorts(4444, 4445)
+            .withEnv("SECRETS_SYSTEM", generateSecret())
+            .withEnv("DSN", DSN)
+            .withEnv("URLS_SELF_ISSUER", "http://hydra:4444/")
+            .withEnv("URLS_CONSENT", "http://consent:3000/consent")
+            .withEnv("URLS_LOGIN", "http://consent:3000/login")
+            .withEnv("OAUTH2_ACCESS_TOKEN_STRATEGY", "jwt")
+            .withEnv("TTL_ACCESS_TOKEN", TTL_ACCESS_TOKEN_IN_SECONDS + "s")
+            .withCommand("serve all --dangerous-force-http")
+            .waitingFor(Wait.forHttp("/health/ready").forPort(4444).forStatusCode(200));
+
+    private final AutoCloseableCloser closer = AutoCloseableCloser.create();
+
+    TestingHydraService()
+    {
+        closer.register(network);
+        closer.register(databaseContainer);
+        closer.register(migrationContainer);
+        closer.register(consentContainer);
+        closer.register(hydraContainer);
+    }
+
+    public void start()
+    {
+        databaseContainer.start();
+        migrationContainer.start();
+        consentContainer.start();
+        hydraContainer.start();
+    }
+
+    public FixedHostPortGenericContainer<?> createHydraContainer()
+    {
+        return new FixedHostPortGenericContainer<>(HYDRA_IMAGE).withNetwork(network);
+    }
+
+    public Network getNetwork()
+    {
+        return network;
+    }
+
+    public int getHydraPort()
+    {
+        return hydraContainer.getMappedPort(4444);
+    }
+
+    @Override
+    public void close()
+            throws IOException
+    {
+        try {
+            closer.close();
+        }
+        catch (Exception e) {
```

Comment:
Change this class to implement `AutoCloseable`, then this catch block is not needed

Reply:
Changing it to `AutoCloseable` makes harder to use it in our case which involves: `Closeables.closeAll(server, testingHydraService)`.

Reply:
`Closeables` supports `AutoCloseable` now

Reply:
That's right, changed.